src/measure_theory/prod_group.lean

```diff
@@ -0,0 +1,190 @@
+  conv_rhs { rw [← map_prod_mul_inv_eq hμ hν] },
+  symmetry, exact lintegral_map hf ((measurable_snd.mul measurable_fst).prod_mk measurable_fst.inv)
+end
+
+lemma measure_mul_right_null (hμ : is_mul_left_invariant μ) {E : set G} (hE : is_measurable E)
+  (y : G) : μ ((λ x, x * y) ⁻¹' E) = 0 ↔ μ E = 0 :=
+begin
+  rw [← measure_inv_null hμ hE, ← hμ y⁻¹ (measurable_inv hE),
+    ← measure_inv_null hμ (measurable_mul_right y hE)],
+  convert iff.rfl using 3, ext x, simp,
+end
+
+lemma measure_mul_right_ne_zero (hμ : is_mul_left_invariant μ) {E : set G} (hE : is_measurable E)
+  (h2E : μ E ≠ 0) (y : G) : μ ((λ x, x * y) ⁻¹' E) ≠ 0 :=
+(not_iff_not_of_iff (measure_mul_right_null hμ hE y)).mpr h2E
+
+/-- A technical lemma relating two different measures. This is basically [Halmos, §60 Th. A].
+  There seems to be a gap in the last step of the proof in [Halmos].
+  In the last line, the equality `g(x⁻¹)ν(Ex⁻¹) = f(x)` holds if we can prove that
+  `0 < ν(Ex⁻¹) < ∞`. The first inequality follows from §59, Th. D, but I couldn't find the second
+  inequality. For this reason, we use a compact `E` instead of a measurable `E` as in [Halmos], and
+  additionally assume that `ν` is a regular measure (we only need that it is finite on compact
+  sets). -/
+lemma measure_lintegral_div_measure [t2_space G] (hμ : is_mul_left_invariant μ)
+  (hν : is_mul_left_invariant ν) (h2ν : regular ν) {E : set G} (hE : is_compact E) (h2E : ν E ≠ 0)
+  (f : G → ennreal) (hf : measurable f) :
+  μ E * ∫⁻ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E) ∂ν = ∫⁻ x, f x ∂μ :=
+begin
+  let Em := hE.is_measurable,
+  symmetry,
+  set g := λ y, f y⁻¹ / ν ((λ h, h * y⁻¹) ⁻¹' E),
+  have hg : measurable g := (hf.comp measurable_inv).ennreal_div
+    ((measurable_measure_mul_right Em).comp measurable_inv),
+  simp_rw [← set_lintegral_one, ← lintegral_indicator _ Em,
+    ← lintegral_lintegral_mul (measurable_const.indicator Em) hg],
+  rw [← lintegral_lintegral_mul_inv hμ hν],
+  have mE : ∀ x : G, measurable (λ y, ((λ z, z * x) ⁻¹' E).indicator (λ z, (1 : ennreal)) y) :=
+  λ x, measurable_const.indicator (measurable_mul_right _ Em),
+  have : ∀ x y, E.indicator (λ (z : G), (1 : ennreal)) (y * x) =
+    ((λ z, z * x) ⁻¹' E).indicator (λ (b : G), 1) y,
+  { intros x y, symmetry, convert indicator_comp_right (λ y, y * x), ext1 z, simp },
+  have h3E : ∀ y, ν ((λ x, x * y) ⁻¹' E) ≠ ⊤ :=
+  λ y, ennreal.lt_top_iff_ne_top.mp (h2ν.lt_top_of_is_compact $
+    (homeomorph.mul_right _).compact_preimage.mpr hE),
+  simp_rw [this, lintegral_mul_const _ (mE _), lintegral_indicator _ (measurable_mul_right _ Em),
+    set_lintegral_one, g, inv_inv,
+    ennreal.mul_div_cancel' (measure_mul_right_ne_zero hν Em h2E _) (h3E _)],
+  exact ((measurable_const.indicator Em).comp measurable_fst).ennreal_mul (hg.comp measurable_snd)
+end
+
+/-- This is roughly the uniqueness (up to a scalar) of left invariant Borel measures on a second
+  countable locally compact group. -/
```

Comment:
Can you give in this docstring a pointer to the lemma giving exactly the uniqueness of Haar measure, proved in the other file?